hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java

```diff
@@ -131,61 +138,115 @@
 
   // NOTE: Increment tests were moved to their own class, TestIncrementsFromClientSide.
   private static final Logger LOG = LoggerFactory.getLogger(TestFromClientSide.class);
-  protected final static HBaseTestingUtility TEST_UTIL = new HBaseTestingUtility();
+  protected static HBaseTestingUtility TEST_UTIL;
   private static byte [] ROW = Bytes.toBytes("testRow");
   private static byte [] FAMILY = Bytes.toBytes("testFamily");
   private static final byte[] INVALID_FAMILY = Bytes.toBytes("invalidTestFamily");
   private static byte [] QUALIFIER = Bytes.toBytes("testQualifier");
   private static byte [] VALUE = Bytes.toBytes("testValue");
   protected static int SLAVES = 3;
 
-  @Rule
-  public TestName name = new TestName();
+  @Rule public TestTableName name = new TestTableName();
+
+  // To keep the child classes happy.
+  TestFromClientSide() {}
 
-  protected static final void initialize(Class<?>... cps) throws Exception {
+  public TestFromClientSide(Class registry, int numHedgedReqs) throws Exception {
+    initialize(registry, numHedgedReqs, MultiRowMutationEndpoint.class);
+  }
+
+  @Parameterized.Parameters
+  public static Collection parameters() {
+    return Arrays.asList(new Object[][] {
+        { MasterRegistry.class, 1},
+        { MasterRegistry.class, 2},
+        { ZKConnectionRegistry.class, 1}
+    });
+  }
+
+  /**
+   * JUnit does not provide an easy way to run a method after each parameterized run. Without that
```

Comment:
No fun, this solution.

Looks like junit-4.13 adds `@{Before,After}Param` method annotations, via junit-team/junit4#1435 . Does this feature work for you?

Reply:
https://issues.apache.org/jira/browse/HBASE-23664

Reply:
Doesn't work. We are on 4.12.